src/main/java/springdata/EmailServiceImpl.java

```diff
@@ -0,0 +1,32 @@
+package springdata;
+
+import exception.MessagingRuntimeException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.mail.javamail.MimeMessageHelper;
+import org.springframework.stereotype.Service;
+
+import javax.mail.MessagingException;
+import javax.mail.internet.MimeMessage;
+
+@Service
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class EmailServiceImpl implements EmailService {
+
+    private final JavaMailSender emailSender;
+
+    @Override
+    public void send(String to, String subject, String text) {
+        MimeMessage message = this.emailSender.createMimeMessage();
+        MimeMessageHelper helper = new MimeMessageHelper(message);
+        try {
+            helper.setTo(to);
+            helper.setSubject(subject);
+            helper.setText(text);
+            this.emailSender.send(message);
+        } catch (MessagingException messageException) {
+            throw new MessagingException(messageException);
```

Comment:
What is the idea of this code? Please explain. You should not catch this exception or if catch it log and throw not wrapping it into the same exception.